### ISSUE TITLE
2.x Routes [method] option no longer working

### ISSUE THREAD

**Issue by challgren** [cakephp/cakephp#15179]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 2.10.22
* Platform and Target: Apache/2.4.46 (cPanel) OpenSSL/1.1.1h mod_bwlimited/1.4 mpm-itk/2.4.7-04 mod_cpanel/1.4

### What you did
My Config/routes.php file is as follows

```php
Router::parseExtensions('json');

Router::connect('/', array('controller' => 'home', 'action' => 'index'));
Router::connect('/robots.txt', array('controller' => 'home', 'action' => 'robots'));
Router::connect('/how-it-works', array('controller' => 'home', 'action' => 'how_it_works'));
Router::connect('/test_version', array('controller' => 'home', 'action' => 'test_version'));
Router::connect('/privacy-policy', array('controller' => 'home', 'action' => 'privacy_policy'));
Router::connect('/faq', array('controller' => 'home', 'action' => 'faq'));
Router::connect('/terms-and-conditions', array('controller' => 'home', 'action' => 'terms_and_conditions'));
Router::connect('/add_business', array('controller' => 'home', 'action' => 'add_business'));
Router::connect('/:prefix/create_account/:id', array('controller' => 'signup', 'action' => 'register'), array('pass' => array(
        'id'), 'id' => '[a-z0-9]+'));

Router::connect('/invited/:id', array('controller' => 'invited', 'action' => 'index'), array('pass' => array('id'), 'id' => '[ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789]+'));
Router::connect('/earn_points/:action', array('controller' => 'EarnPoints'));
Router::connect('/signup/:action', array('controller' => 'Signup'));
Router::connect('/signup/:action/:id', array('controller' => 'Signup'), array('pass' => array('id')));
Router::connect('/welcome/:action', array('controller' => 'Welcome'));

Router::connect('/place/:id/info_window', array('controller' => 'Place', 'action' => 'info_window'), array('pass' => array('id')));
Router::connect('/place/:id/:slug', array('controller' => 'Place', 'action' => 'view'), array('pass' => array('id')));

Router::connect('/places/list', array('controller' => 'Places', 'action' => 'list_places'));
Router::connect('/places/:action', array('controller' => 'Places'));
Router::connect('/:prefix/places/:action', array('controller' => 'Places'));

Router::connect('/:prefix/reset_password/:id', array('controller' => 'ResetPassword'), array('pass' => array('id')));

Router::connect('/:prefix/verify/:id', array('controller' => 'Verify'), array('pass' => array('id')));

Router::mapResources(array(':controller'));
foreach (Router::$_resourceMap as $params) {
    $url = '/:prefix/:controller' . (($params['id']) ? '/:id' : '');
    Router::connect($url, array(
        'action' => $params['action'],
        '[method]' => $params['method']
            ), array('id' => static::ID . '|' . static::UUID, 'pass' => array('id'))
    );
}
Router::connect('/:prefix/card/:id/add_points', array(
    'controller' => 'card',
    'action' => 'add_points',
    '[method]' => 'PUT'
        ), array('pass' => array('id'))
);
Router::connect('/:prefix/card/:id/redeem_amount', array(
    'controller' => 'card',
    'action' => 'redeem_amount',
    '[method]' => 'PUT'
        ), array('pass' => array('id'))
);
Router::connect('/:prefix/card_transactions/:id/pos_transactions', array(
    'controller' => 'card_transactions',
    'action' => 'pos_transactions',
    '[method]' => 'POST'
        ), array('pass' => array('id'))
);

Router::connect(
        '/swagger/docs', array('controller' => 'Swagger')
);

/**
 * Load all plugin routes. See the CakePlugin documentation on
 * how to customize the loading of plugin routes.
 */
CakePlugin::routes();

/**
 * Load the CakePHP default routes. Only remove this if you do not want to use
 * the built-in default routes.
 */
require CAKE . 'Config' . DS . 'routes.php';
```

### What happened
The route for PUT /:prefix/card/:id/add_points and /:prefix/card/:id/redeem_amount is failing with not found. 

```
2020-11-19 12:34:12 Error: [MissingActionException] Action CardController::pos_619() could not be found.
Exception Attributes: array (
  'controller' => 'CardController',
  'action' => 'pos_619',
)
Request URL: /pos/card/619/add_points.json
Stack Trace:
#0 /Vendor/cakephp/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#1 /Vendor/cakephp/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(CardController), Object(CakeRequest))
#2 /public_html/webroot/index.php(98): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#3 {main}
```

### What you expected to happen
The route to be connected just fine as this worked up until about start of November. My only solution is to change the `[method]` to `method` and it works just fine. But then the route below fails to be connected.

```php
Router::mapResources(array(':controller'));
foreach (Router::$_resourceMap as $params) {
    $url = '/:prefix/:controller' . (($params['id']) ? '/:id' : '');
    Router::connect($url, array(
        'action' => $params['action'],
        'method' => $params['method']
            ), array('id' => static::ID . '|' . static::UUID, 'pass' => array('id'))
    );
}
```

**Comment by challgren** [comment]
Digging deeper into this, it seems like $_REQUEST isn't being populated. Further debugging with `$_SERVER['REQUEST_METHOD']` show it as a GET request, when its a PUT request. Also query parameters aren't being populated either.

**Comment by challgren** [comment]
Ok so this issue is only occurring with PUT requests

**Comment by challgren** [comment]
Ok so this can be closed due to mod_security having a rule that limits PUT DELETE and PATCH requests.

**Comment by othercorey** [comment]
> The route to be connected just fine as this worked up until about start of November.

Can you document what changed for users who search?

**Comment by challgren** [comment]
@othercorey cPanel must have did a change to the `OWASP CRS v4.x for ModSec 2.9` rules and added a new rule `9111100` that prevents PUT, PATCH, DELETE requests. After going into WHM and clicking on ModSecurity Tools I noticed a bunch of log entries that had `911100: Method is not allowed by policy` after clicking on the edit button for the Rule ID, I unchecked the `Enable Rule` box and checked the `Deploy and Restart Apache` button and this stopped enforcement of the rule and restarted apache. Everything is working fine now.
